alphastats/gui/utils/analysis.py

```diff
@@ -0,0 +1,155 @@
+            + [custom_group_option],
+        )
+
+        column = None
+        if grouping_variable == default_option:
+            st.stop()  # TODO: using stop here is not really great
+        elif grouping_variable != custom_group_option:
+            unique_values = metadata[grouping_variable].unique().tolist()
+
+            column = grouping_variable
+            group1 = st.selectbox("Group 1", options=unique_values)
+            group2 = st.selectbox("Group 2", options=list(reversed(unique_values)))
+
+        else:
+            group1 = st.multiselect(
+                "Group 1 samples:",
+                options=metadata[Cols.SAMPLE].to_list(),
+            )
+
+            group2 = st.multiselect(
+                "Group 2 samples:",
+                options=list(reversed(metadata[Cols.SAMPLE].to_list())),
+            )
+
+            intersection_list = list(set(group1).intersection(set(group2)))
+            if len(intersection_list) > 0:
+                st.warning(
+                    "Group 1 and Group 2 contain same samples: "
+                    + str(intersection_list)
+                )
+
+        if group1 == group2:
+            st.error(
+                "Group 1 and Group 2 can not be the same. Please select different groups."
+            )
+            st.stop()
+
+        self._parameters.update({"group1": group1, "group2": group2})
+        if column is not None:
+            self._parameters["column"] = column
+
+
+class VolcanoPlotAnalysis(GroupCompareAnalysis):
+    """Widget for Volcano Plot analysis."""
+
+    def show_widget(self):
+        """Show the widget and gather parameters."""
+        super().show_widget()
+
+        parameters = {}
+        method = st.selectbox(
+            "Differential Analysis using:",
+            options=["ttest", "anova", "wald", "sam", "paired-ttest", "welch-ttest"],
+        )
+        parameters["method"] = method
+
+        parameters["labels"] = st.checkbox("Add labels", value=True)
+
+        parameters["draw_line"] = st.checkbox("Draw lines", value=True)
+
+        parameters["alpha"] = st.number_input(
+            label="alpha", min_value=0.001, max_value=0.050, value=0.050
+        )
+
+        parameters["min_fc"] = st.select_slider(
+            "Foldchange cutoff", range(0, 3), value=1
+        )
+
+        if method == "sam":
+            parameters["perm"] = st.number_input(
+                label="Number of Permutations", min_value=1, max_value=1000, value=10
+            )
+            parameters["fdr"] = st.number_input(
+                label="FDR cut off", min_value=0.005, max_value=0.1, value=0.050
+            )
+
+        self._parameters.update(parameters)
+
+    def do_analysis(self):
+        """Draw Volcano Plot using the VolcanoPlot class.
+
+        Returns a tuple(figure, analysis_object, parameters) where figure is the plot,
+        analysis_object is the underlying object, parameters is a dictionary of the parameters used.
```

Comment:
Not sure if this will be addessed in the other unreviewed PRs, but I would think a split into calculating analysis and plotting results.